addons/mrp/models/mrp_workorder.py

```diff
@@ -271,9 +271,11 @@ def _compute_barcode(self):
             wo.barcode = f"{wo.production_id.name}/{wo.id}"
 
     @api.depends('production_id', 'product_id')
+    @api.depends_context('within_gantt')
     def _compute_display_name(self):
+        fct = (lambda w: f"{w.product_id.name} - {w.production_id.name} - {w.name}") if self.env.context.get('within_gantt') else (lambda w: f"{w.production_id.name} - {w.name}")
         for wo in self:
-            wo.display_name = f"{wo.production_id.name} - {wo.name}"
+            wo.display_name = fct(wo)
```

Comment:
```suggestion
            wo.display_name = f"{wo.production_id.name} - {wo.name}"
            if self.env.context.get('within_gantt'):
                wo.display_name = f"{w.product_id.name} - {w.production_id.name} - {w.name}"
```
It seems a bit more readable :)

Reply:
I would add a more explicit context like prefix_product

Reply:
I thought it was an elegant pythonic way to avoid n-1 unnecessary tests,
done